Apps/W1/Shopify/app/src/Products/Codeunits/ShpfyProductPriceCalc.Codeunit.al

```diff
@@ -91,22 +92,37 @@ codeunit 30182 "Shpfy Product Price Calc."
     /// Create Temp Sales Header.
     /// </summary>
     local procedure CreateTempSalesHeader()
+    var
+        CustomerRec: Record Customer;
     begin
         Clear(TempSalesHeader);
         TempSalesHeader."Document Type" := TempSalesHeader."Document Type"::Quote;
         TempSalesHeader."No." := Shop.Code;
-        TempSalesHeader."Sell-to Customer No." := Shop.Code;
-        TempSalesHeader."Bill-to Customer No." := Shop.Code;
+        if CustomerNo <> '' then begin
+            if CustomerRec.GET(CustomerNo) then begin
+                TempSalesHeader."Sell-to Customer No." := CustomerNo;
+                TempSalesHeader."Bill-to Customer No." := CustomerNo;
+                TempSalesHeader."Customer Price Group" := CustomerRec."Customer Price Group";
+                TempSalesHeader."Customer Disc. Group" := CustomerRec."Customer Disc. Group";
+                TempSalesHeader."Allow Line Disc." := CustomerRec."Allow Line Disc.";
+                TempSalesHeader."Invoice Disc. Code" := CustomerRec."Invoice Disc. Code";
+            end
+        end
+        else begin
+            TempSalesHeader."Sell-to Customer No." := Shop.Code;
+            TempSalesHeader."Bill-to Customer No." := Shop.Code;
+            TempSalesHeader."Customer Price Group" := CustomerPriceGroup;
+            TempSalesHeader."Customer Disc. Group" := CustomerDiscGroup;
+            TempSalesHeader."Allow Line Disc." := AllowLineDisc;
+        end;
+
         TempSalesHeader."Gen. Bus. Posting Group" := GenBusPostingGroup;
         TempSalesHeader."VAT Bus. Posting Group" := VATBusPostingGroup;
```

Comment:
Some unit test would be great.

There is a codeunit that already has a test for catalog.
https://github.com/microsoft/ALAppExtensions/blob/main/Apps/W1/Shopify/test/Catalogs/ShpfyCatalogPricesTest.Codeunit.al

I suggest to add 4 more in an equivalent manner, make sure to populate Customer No in the Catalog record.
1
CustomerX
Product1 unit price 100  Item discount category DISCOUNT
General discount for DISCOUNT= 20%
Exported prices: Product1 = 80

2
CustomerX
Product2 unit price 100  Item discount category PAINT
General discount for DISCOUNT= 20%
Customer specific discount for Product2 = 30%
Exported prices: Product2 = 70

3
CustomerX
Product3 unit price 100  Item discount category PAINT
General discount for DISCOUNT= 20%
Product specific discount for Product3 = 25% (for All Customers)
Exported prices: Product3 = 75

4
CustomerX
Product4 unit price 100  Item discount category PAINT
General discount for DISCOUNT= 20%
Customer specific discount for Product4 = 30%, Product specific discount for Product4 = 25%
Exported prices: Product4 = 70.

Reply:
@AndreiPanko I added Test cases. Can you please take a look